universal-application-tool-0.0.1/app/views/AzureFileUploadViewStrategy.java

```diff
@@ -0,0 +1,220 @@
+import views.style.Styles;
+
+public class AzureFileUploadViewStrategy extends FileUploadViewStrategy {
+
+  private static final String AZURE_STORAGE_BLOB_WEB_JAR =
+      "lib/azure__storage-blob/browser/azure-storage-blob.min.js";
+
+  private final ViewUtils viewUtils;
+
+  @Inject
+  AzureFileUploadViewStrategy(ViewUtils viewUtils) {
+    this.viewUtils = viewUtils;
+  }
+
+  @Override
+  public ContainerTag signedFileUploadFields(
+      ApplicantQuestionRendererParams params, FileUploadQuestion fileUploadQuestion) {
+    StorageUploadRequest storageUploadRequest = params.signedFileUploadRequest().get();
+    if (!(storageUploadRequest instanceof BlobStorageUploadRequest)) {
+      throw new RuntimeException(
+          "Trying to upload file to Azure blob storage using incorrect" + " upload request type.");
+    }
+    BlobStorageUploadRequest request = (BlobStorageUploadRequest) storageUploadRequest;
+    ContainerTag formTag = form().withId("azure-upload-form-component");
+    formTag.with(
+        footer(viewUtils.makeWebJarsTag(AZURE_STORAGE_BLOB_WEB_JAR)),
+        footer(viewUtils.makeLocalJsTag("azure_upload")));
+    return formTag
+        .with(input().withType("file").withName("file"))
+        .with(input().withType("hidden").withName("sasToken").withValue(request.sasToken()))
+        .with(input().withType("hidden").withName("blobUrl").withValue(request.blobUrl()))
+        .with(
+            input().withType("hidden").withName("containerName").withValue(request.containerName()))
+        .with(input().withType("hidden").withName("fileName").withValue(request.fileName()))
+        .with(input().withType("hidden").withName("accountName").withValue(request.accountName()))
+        .with(
+            input()
+                .withType("hidden")
+                .withName("successActionRedirect")
+                .withValue(request.successActionRedirect()))
+        .with(TagCreator.button(text("Upload to Azure Blob Storage")).withType("submit"));
+  }
+
+  @Override
+  public Tag renderFileUploadBlockSubmitForms(
+      Params params, ApplicantQuestionRendererFactory applicantQuestionRendererFactory) {
+    String key = FileNameFormatter.formatFileUploadQuestionFilename(params);
+    String onSuccessRedirectUrl =
+        params.baseUrl()
+            + routes.ApplicantProgramBlocksController.updateFile(
+                    params.applicantId(),
+                    params.programId(),
+                    params.block().getId(),
+                    params.inReview())
+                .url();
+
+    StorageUploadRequest request =
+        params.storageClient().getSignedUploadRequest(key, onSuccessRedirectUrl);
+
+    if (!(request instanceof BlobStorageUploadRequest)) {
+      throw new RuntimeException(
+          "Tried to upload a file to Azure Blob storage using incorrect request type");
+    }
+    BlobStorageUploadRequest blobStorageUploadRequest = (BlobStorageUploadRequest) request;
```

Comment:
suggestion (blocking): extract a private method that reduces the duplication between here and `signedFileUploadFields` for asserting the type and performing the cast.

Reply:
Done.